moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -108,8 +144,76 @@ export default function MapView({ projectName, projectStatuses }) {
   };
 
   const onSaveComponent = () => {
-    const newComponents = [...components, draftComponent];
-    setComponents(newComponents);
+    /* Start data preparation */
+    const {
+      component_id,
+      description,
+      moped_subcomponents,
+      component_name,
+      internal_table,
+      features,
+    } = draftComponent;
+
+    const subcomponentsArray = moped_subcomponents
+      ? moped_subcomponents.map((subcomponent) => ({
+          subcomponent_id: subcomponent.value,
+        }))
+      : [];
+
+    const featureTable = internal_table;
+
+    const featuresToInsert = [];
+    const drawnLinesToInsert = [];
+    const drawnPointsToInsert = [];
+
+    const drawnFeatures = features.filter((feature) =>
+      Boolean(getDrawId(feature))
+    );
+    const selectedFeatures = features.filter(
+      (feature) => !Boolean(getDrawId(feature))
+    );
+
+    if (featureTable === "feature_street_segments") {
+      makeLineStringFeatureInsertionData(
+        featureTable,
+        selectedFeatures,
+        featuresToInsert
+      );
+      makeDrawnLinesInsertionData(drawnFeatures, drawnLinesToInsert);
+    } else if (
+      featureTable === "feature_intersections" ||
+      featureTable === "feature_signals"
+    ) {
+      makePointFeatureInsertionData(
+        featureTable,
+        selectedFeatures,
+        featuresToInsert
+      );
+      makeDrawnPointsInsertionData(drawnFeatures, drawnPointsToInsert);
+    }
+
+    const newComponentData = {
+      description,
+      component_id,
+      name: component_name,
+      project_id: projectId,
+      moped_proj_components_subcomponents: {
+        data: subcomponentsArray,
+      },
+      [featureTable]: {
+        data: featuresToInsert,
+      },
+      feature_drawn_lines: { data: drawnLinesToInsert },
+      feature_drawn_points: { data: drawnPointsToInsert },
+    };
+    /* End data preparation */
+
+    addProjectComponent({ variables: { object: newComponentData } }).then(
+      () => {
+        refetchProjectComponents();
+      }
+    );
```

Comment:
Cool, I'll add a log here for now. It would be great to have some feedback like "Try again" or something. Good for a backlog issue, and we might be able to pull an idea from Rebecca's designs IIRC.